contracts/GnosisSafe.sol

```diff
@@ -62,330 +16,7 @@ contract GnosisSafe {
-            require(newContract != 0);
-            ContractCreation(newContract);
-        }
-    }
-
-    function executeCall(address to, uint256 value, bytes data)
-        internal
-        returns (bool success)
-    {
-        assembly {
-            success := call(not(0), to, value, add(data, 0x20), mload(data), 0, 0)
-        }
-    }
-
-    function executeDelegateCall(address to, bytes data)
-        internal
-        returns (bool success)
-    {
-        assembly {
-            success := delegatecall(not(0), to, add(data, 0x20), mload(data), 0, 0)
-        }
-    }
-
-    function executeCreate(bytes data)
-        internal
-        returns (address newContract)
-    {
-        assembly {
-            newContract := create(0, add(data, 0x20), mload(data))
-        }
-    }
-
-    /// @dev Returns transactions hash to be signed by owners.
-    /// @param to Destination address.
-    /// @param value Ether value.
-    /// @param data Data payload.
-    /// @param operation Operation type.
-    /// @param _nonce Transaction nonce.
-    /// @return Transaction hash.
-    function getTransactionHash(address to, uint256 value, bytes data, Operation operation, uint256 _nonce)
-        public
-        view
-        returns (bytes32)
-    {
-        return keccak256(byte(0x19), this, to, value, data, operation, _nonce);
-    }
-
-    /// @dev Returns array of owners.
-    /// @return Array of Safe owners.
-    function getOwners()
-        public
-        view
-        returns (address[])
-    {
-        return owners;
-    }
-
-    /// @dev Returns array of extensions.
-    /// @return Array of extensions.
-    function getExtensions()
-        public
-        view
-        returns (Extension[])
-    {
-        return extensions;
-    }
-
-    /// @dev Returns a the count of owners that have confirmed the given transaction.
-    /// @param transactionHash Safe transaction hash.
-    function getConfirmationCount(bytes32 transactionHash)
-        public
-        view
-        returns (uint confirmationCount)
-    {
-        for (uint i = 0; i < owners.length; i++) {
-            if (isConfirmed[owners[i]][transactionHash])
-                confirmationCount++;
-        }
-    }
-
-    /// @dev Returns a list of owners that have confirmed the given transaction.
-    /// @param transactionHash Safe transaction hash.
-    function getConfirmingOwners(bytes32 transactionHash)
-        public
-        view
-        returns (address[] confirmingOwners)
-    {
-        uint confirmationCount = getConfirmationCount(transactionHash);
-        confirmingOwners = new address[](confirmationCount);
-        confirmationCount = 0;
-        for (uint i = 0; i < owners.length; i++) {
-            if (isConfirmed[owners[i]][transactionHash]) {
-                confirmingOwners[confirmationCount] = owners[i];
-                confirmationCount++;
-            }
-        }
+        setupOwners(_owners, _threshold);
+        setupModules(to, data);
```

Comment:
We could make setupOwners and setupModules internal functions. They can still be called in the Safe setup function and we won't have to set the initialized storage in the modules manager.

Reply:
I made setupModules public to have the possibility to just deploy the modulemanager (that was more the case for personal edition module solution). I can take that out and if somebody wants this functionality they have to add a `StandaloneModuleManger` that has this functionality